src/DependencyResolver/Package.php

```diff
@@ -25,7 +26,10 @@
  */
 final class Package
 {
-    /** @param list<ConfigureOption> $configureOptions */
+    /**
+     * @param list<ConfigureOption> $configureOptions
+     * @param list<string>          $excludedOsFamilies
```

Comment:
should be a `list<OsFamily>` and create a new `enum` for this with the allowed values:

 * Windows
 * BSD
 * Darwin
 * Solaris
 * Linux
 * Unknown

ref: https://www.php.net/manual/en/reserved.constants.php#constant.php-os-family